internal/cache/file/downloader/parallel_downloads_job.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downloader
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/cache/lru"
+	cacheutil "github.com/googlecloudplatform/gcsfuse/v2/internal/cache/util"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/logger"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/monitor"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/util"
+	"golang.org/x/sync/errgroup"
+)
+
+// downloadRangeToFile is a helper function to download a given range of object
+// from GCS into given file handle.
+//
+// This function doesn't take locks and can be executed parallely.
+func (job *Job) downloadRangeToFile(ctx context.Context, cacheFile *os.File, start, end int64) error {
```

Comment:
Should we pass the offsetWriter here?

Reply:
Trying to offload some work from `parallelDownloadObjectAsync` as that's already long and allow unit testing that separately.

Reply:
Done. PTAL